.circleci/config.yml

```diff
@@ -3,45 +3,119 @@
+        - v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
+        # fallback to using the latest cache if no exact match is found
+        - v1-dependencies-
 
-      # Download and cache dependencies
-      - restore_cache:
-          keys:
-          - v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
+  restore_cache_python36: &restore_cache_python36
+    restore_cache:
+      keys:
+        - v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
+        # fallback to using the latest cache if no exact match is found
+        - v1-dependencies-
+  
+  save_cache_python27: &save_cache_python27
+    save_cache:
+      paths:
+        - ~/venv27
+      key: v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
 
+  save_cache_python36: &save_cache_python36
+    save_cache:
+      paths:
+        - ~/venv36
+      key: v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
+
+jobs:
+  build_python27:
+    <<: *container_python27
+    steps:
+      - checkout
+      - *restore_cache_python27
       - run:
           name: install dependencies
           command: |
             pip install virtualenv
-            virtualenv venv
-            . venv/bin/activate
+            virtualenv ~/venv27
+            . ~/venv27/bin/activate
             pip install -r requirements.txt
             pip install -r requirements-dev.txt
+            pip install .
+      - *save_cache_python27
+      - run:
+          name: run tests
+          environment:
+            LOCALSTACK_HOST: localstack
+          command: |
+            . ~/venv27/bin/activate
+            CUMULUS_ENV=testing nosetests -v -s
 
-      - save_cache:
-          paths:
-            - ./venv
-          key: v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}
-        
+  build_python36:
+    <<: *container_python36
+    steps:
+      - checkout
+      - *restore_cache_python36
+      - run:
+          name: install dependencies
+          command: |
+            pip install --user virtualenv
+            virtualenv ~/venv36
+            . ~/venv36/bin/activate
+            pip install -r requirements.txt
+            pip install -r requirements-dev.txt
+            pip install .
+      - *save_cache_python36
       - run:
           name: run tests
           environment:
             LOCALSTACK_HOST: localstack
           command: |
-            . venv/bin/activate
+            . ~/venv36/bin/activate
             CUMULUS_ENV=testing nosetests -v -s
 
-          
\ No newline at end of file
+  publish_pypi:
+    <<: *container_python36
+    steps:
+      - checkout
+      - *restore_cache_python36
+      - run:
+          name: Deploy to PyPi
+          command: |
```

Comment:
In the `publish_pypi` `command` there should also be a check to see if the current version has been published, and if so, skip publishing.

It didn't have this in cumulus-message-adapter because it was checking against github releases to see if the current version had been published, but we're not doing github releases here.

Reply:
After some research, found that we don't need this step because PyPI will never let us upload an existing version, it always forces you to bump the version.